### DISCUSSION TITLE
Traffic Layer

### DISCUSSION THREAD

**Discussion by hdrodel** [inocan-group/vue3-google-map#274] [Q&A] [answered]
What would be the best practice to display a traffic layer on the map?

**Comment by hdrodel** [comment]
I got something working but i don't know if that really is how it was supposed to be done
```
<template>
<GoogleMap
   api-key="<key>"
   :api="google"
   ref="mapRef"
    > 
    <!--adding markers here-->
</GoogleMap>
</template>
<script>
import { Loader } from "@googlemaps/js-api-loader";
export default {
  data() {
    return {
    google: null,
    loader: new Loader({apiKey: <key>, version: "weekly"})
};
  },

  async mounted() {
    this.google = await this.loader.importLibrary("maps");
    this.trafficLayer = new this.google.TrafficLayer();
    await nextTick();
    console.log(this.trafficLayer);
    this.trafficLayer.setMap(this.$refs.mapRef.map);
},

};
</script>
```

Any better way to do this?

**Comment by HusamElbashir** [comment, marked as answer]
As we don't have a Traffic Layer component your only option is to set things up manually like you did. You can use the `libraries` prop on `GoogleMap` which should simplify the loading of the script with the required library for you. Refer to the [Advanced Usage](https://github.com/inocan-group/vue3-google-map?tab=readme-ov-file#advanced-usage) section as well for ways to access the API after loading the script.
